client/components/timezone-dropdown/index.jsx

```diff
@@ -3,50 +3,100 @@
  * External Dependencies
  */
 import React from 'react';
-import noop from 'lodash/noop';
+import i18n from 'lib/mixins/i18n';
 
 /**
  * Internal dependencies
  */
+import wpcom from 'lib/wp';
 import Dropdown from 'components/select-dropdown';
-import i18n from 'lib/mixins/i18n';
 
-export default React.createClass( {
-	displayName: 'TimezoneDropdown',
+/**
+ * Module variables
+ */
+const { Component, PropTypes } = React;
+const noop = () => {};
+const undocumented = wpcom.undocumented();
+
+class TimezoneDropdown extends Component {
+	constructor() {
+		super();
 
-	propTypes: {
-		selectedZone: React.PropTypes.string,
-		onSelect: React.PropTypes.func
-	},
+		// bound methods
+		this.onSelect = this.onSelect.bind( this );
 
-	getDefaultProps() {
-		return {
-			onSelect: noop
+		this.state = {
+			timezones: [ { label: '', value: '' } ]
 		};
-	},
-
-	getTimezoneNames() {
-		return i18n.moment.tz.names().map( zone => {
-			return ( {
-				label: zone,
-				value: zone
-			} );
+	}
+
+	componentWillMount() {
+		undocumented.timezones( ( err, zones ) => {
```

Comment:
:-o can we do it in another PR please? 
I promise to make this improvement as soon as it is merged.